src/Composer/ComposerEnvironment.php

```diff
@@ -107,11 +108,25 @@ public function checkFilePermissions()
             $this->getTemporaryVendorPath(),
         ];
 
+        $paths = [];
+
         foreach ($pathsToCheck as $path) {
             if (file_exists($path) && !is_writable($path)) {
-                throw new FilePermissionException('Write permission missing for '.$path);
+                $paths[] = str_replace(base_path(), '/', $path);
             }
         }
+
+        return $paths;
+    }
+
+    /**
+     * @throws FilePermissionException
+     */
+    public function checkFilePermissions()
+    {
+        foreach ($this->retrieveFilePermissions() as $path) {
+            throw new FilePermissionException('Write permission missing for '.$path);
```

Comment:
I think we should still provide the full path to check in the log file even if we truncate it in the UI